src/main/java/org/apache/sysml/runtime/matrix/data/SparseBlockCSR.java

```diff
@@ -848,7 +848,57 @@ public String toString() {
 		
 		return sb.toString();
 	}
-	
+
+//	@Override
+	public boolean checkValidity(int rlen, int clen, boolean strict) {
+		//1. correct meta data
+		if( rlen < 0 || clen < 0 ) {
+			throw new RuntimeException("Invalid block dimensions: "+rlen+" "+clen);
+		}
+
+		//2. correct array lengths
+		int nnz = _size;
+		long esize = estimateMemory(rlen, clen, MatrixBlock.SPARSITY_TURN_POINT);
+		if(_size < esize && _ptr.length > rlen+1 && _values.length  >= nnz && _indexes.length >= nnz ) {
+			throw new RuntimeException("Incorrect array lengths.");
+		}
+
+		//3. non-decreasing row pointers
+		for( int i=1; i<rlen; i++ ) {
+			if(_ptr[i-1] > _ptr[i])
+				throw new RuntimeException("Row pointers are decreasing.");
+		}
+
+		//4. sorted column indexes per row
+		for( int i=0; i<rlen; i++ ) {
+			int apos = _ptr[i];
+			int alen = _size;
```

Comment:
please call pos(i) and size(i) instead, because size would be incorrect (all nnz of the block).